assessment/libs/shapes.php

```diff
@@ -0,0 +1,1465 @@
+        $args = $args . "strokewidth=2;arc([{$arcStartDisp[0]},{$arcStartDisp[1]}],[{$arcEndDisp[0]},{$arcEndDisp[1]}]),$rArc[$j];strokewidth=1;";
+      }
+    }
+    // Draw arc(s) for angle 2
+    if (is_numeric($arcNum[$arcType[2]])) {
+      for ($j=0;$j<$arcNum[$arcType[2]];$j++) {
+        $arcStart = [$xf[1] - $rArc[$j]*cos($angRad[0]),$yf[1] - $rArc[$j]*sin($angRad[0])];
+        $arcEnd = [$xf[1] + $rArc[$j]*cos($angRad[1]),$yf[1] - $rArc[$j]*sin($angRad[1])];
+        $arcStartDisp = [cos($getBack)*$arcStart[0]-sin($getBack)*$arcStart[1],sin($getBack)*$arcStart[0]+cos($getBack)*$arcStart[1]];
+        $arcEndDisp = [cos($getBack)*$arcEnd[0]-sin($getBack)*$arcEnd[1],sin($getBack)*$arcEnd[0]+cos($getBack)*$arcEnd[1]];
+        $args = $args . "strokewidth=2;arc([{$arcStartDisp[0]},{$arcStartDisp[1]}],[{$arcEndDisp[0]},{$arcEndDisp[1]}]),$rArc[$j];strokewidth=1;";
+      }
+    }
+  }
+  $gr = showasciisvg("setBorder(10);initPicture($xminDisp,$xmaxDisp,$yminDisp,$ymaxDisp);$args;",$size,$size);
+  
+  return $gr;
+}
+
+//--------------------------------------------draw_polygon()----------------------------------------------------
+
+// draw_polygon([number],[options])
+// draw_polygon() will draw a random polygon from 3 to 9 sides.
+// draw_polygon(n) will draw a polygon with n sides.
+// Options include:
+// "points,[lab1,lab2,...]" Draws points on the vertices with optional labels for vertices
+// "regular" Makes the polygon regular.
+// "norotate" Makes the first vertex on the positive x-axis.
+
+function draw_polygon() {
+  
+  $randSides = rand(3,9);
+  $isRegular = false;
+  $hasPoints = false;
+  $hasSides = false;
+  $noRotate = false;
+  $size = 300;
+  $rotatePolygon = false;
+  
+  $input = func_get_args();
+  $argsArray = [];
+  foreach ($input as $list) {
+    if (!is_array($list)) {
+      $list = listtoarray($list);
+    }
+    $list = array_map('trim', $list);
+    $argsArray[] = $list;
+  }
+  if (count($argsArray) == 0) {
+    $argsArray[] = [$randSides];
+  }
+  if (count($argsArray) > 0 && !is_numeric($argsArray[0][0])) {
+    echo "Warning! First input must be a whole number greater than 2.";
+    array_unshift($argsArray,[$randSides]);
+  }
+  
+  foreach ($argsArray as $key => $in) {
+    if ($in[0]=="sides") {
+      $hasSides = true;
+      $sideKey = $key;
+    }
+    if (in_array("rotate",$in)) {
+      $rotatePolygon = true;
+    }
+    if ($in[0]=="size" && isset($in[1])) {
+      $size = $in[1];
+    }
+    if (in_array("regular",$in)) {
+      $isRegular = true;
+    }
+    if ($in[0]=="points") {
+      $hasPoints = true;
+      $pointKey = $key;
+    }
+    if (in_array("axes",$in)) {
+      $args = "stroke='grey';line([-1.3,0],[1.3,0]);line([0,-1.3],[0,1.3]);stroke='black';";
+    }
+  }
+  
+  //number of sides
+  $n = $argsArray[0][0];
+  if (!is_numeric($n) || abs($n-round($n))>1E-9 || $n < 3) {
```

Comment:
Small request: can you add a max value for $n too, to prevent the loops from getting too outrageous?  Maybe a max of 50 sides or something?